src/main/java/com/aws/greengrass/mqttclient/MqttClient.java

```diff
@@ -468,80 +500,112 @@ private void triggerSpooler() {
      * Unsubscribe from a MQTT topic.
      *
      * @param request unsubscribe request
-     * @throws ExecutionException   if an error occurs
-     * @throws InterruptedException if the thread is interrupted while unsubscribing
-     * @throws TimeoutException     if the request times out
+     * @throws MqttRequestException if the request is invalid for any reason
      */
-    public synchronized void unsubscribe(UnsubscribeRequest request)
-            throws ExecutionException, InterruptedException, TimeoutException {
+    public synchronized CompletableFuture<Void> unsubscribe(Unsubscribe request) throws MqttRequestException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.